src/search/searcher.h

```diff
@@ -343,18 +347,21 @@ class Searcher : public std::enable_shared_from_this<Searcher> {
         }
 
         const auto ttMove = tryFetchTtMove(hashProbe);
+        const auto prevMove = m_stackItr == m_stack.begin() ? std::nullopt : std::optional<movegen::Move>((m_stackItr - 1)->move);
 
         const auto phase = tbMoves ? PickerPhase::Syzygy : PickerPhase::TtMove;
 
         MovePicker picker(phase);
 
-        while (const auto moveOpt = picker.pickNextMove(board, m_searchTables, moves, m_ply, ttMove)) {
+        while (const auto moveOpt = picker.pickNextMove(board, m_searchTables, moves, m_ply, ttMove, prevMove)) {
             const auto move = moveOpt.value();
 
             if (!makeMove(board, move)) {
                 continue;
             }
 
+            m_stackItr->move = move;
```

Comment:
should probably go into `makeMove` where the other stack values are set

Reply:
I'm not sure I agree. We only use the move for the counter moves, which are only updated inside negamax. So for qsearch we'd be adding moves to the stack and removing them, while knowing they'd never currently be used.

Reply:
> I'm not sure I agree. We only use the move for the counter moves, which are only updated inside negamax. So for qsearch we'd be adding moves to the stack and removing them, while knowing they'd never currently be used.

we never remove stack information. We just decrement the iterator. The memory is already allocated, so there's no overhead, as we always update the stack when incrementing the iterator.

I think it's error prone to update the stackItr in multiple locations. The only reason we don't update the `static_eval` is because it's extremely expensive if we were not to use it. But copying 16bits is literally free :)